### PR TITLE
Lazy load/sysmodule

### PR DESCRIPTION
**Pull request by ckurtz22**
You'll probably want to change the title ID I just set it to something random.
Build process can probably be cleaned up too but I'm not that good with that stuff.

### CODE REVIEW

File: include/SwitchMtpDatabase.h [hunk 1]
```diff
@@ -106,6 +107,7 @@ class SwitchMtpDatabase : public android::MtpDatabase {
                 entry.path = p.string();
                 entry.object_format = MTP_FORMAT_ASSOCIATION;
                 entry.object_size = 0;
+                entry.scanned = false;
```
Comment (m4xw) [inline include/SwitchMtpDatabase.h:110]: Add \n afterwards

File: include/SwitchMtpDatabase.h [hunk 2]
```diff
@@ -154,6 +155,11 @@ class SwitchMtpDatabase : public android::MtpDatabase {
         {
             add_file_entry(*it, parent, storage);
         }
+        try {
```
Comment (m4xw) [inline include/SwitchMtpDatabase.h]: Remove try catch, add sanity check

File: mtp-server-nx.json [hunk 3]
```diff
@@ -0,0 +1,172 @@
+{
+	"name":	"mtp-server-nx",
+	"title_id":	"0x2200000000000200",
```
Comment (m4xw) [inline mtp-server-nx.json]: Use 00FF00006D7470FF

File: source/main.cpp [hunk 4]
```diff
@@ -29,8 +29,68 @@
 
 using namespace android;
 
+// #define APPLET
```
Comment (m4xw) [inline source/main.cpp]: Default to applet

File: source/main.cpp [hunk 5]
```diff
@@ -42,6 +102,9 @@ static void stop_thread(MtpServer* server)
             break;
         }
     }
+    #else
+    serverExit = server;
+    #endif
```
Comment (m4xw) [inline source/main.cpp]: #endif // WANT_APPLET
Also refactor it to WANT_APPLET and WANT_SYSMODULE

File: source/main.cpp [hunk 6]
```diff
@@ -71,8 +134,10 @@ int main(int argc, char* argv[])
         }
     }
 
+#ifdef APPLET
     consoleInit(NULL);
     std::cout << "Press + to exit";
+#endif
```
Comment (m4xw) [inline source/main.cpp]: // WANT_APPLET

**Review by m4xw — CHANGES_REQUESTED** [review]

File: source/main.cpp [hunk 7]
```diff
@@ -29,8 +29,69 @@
 
 using namespace android;
 
+// Pick one
+// #define WANT_APPLET
+#define WANT_SYSMODULE
```
Comment (m4xw) [inline source/main.cpp]: Default to applet

Comment (ckurtz22) [reply source/main.cpp]: Oof. I changed that and then wanted to test the try catch fix and forgot to change back again lol

File: source/main.cpp [hunk 8]
```diff
@@ -29,8 +29,66 @@
 
 using namespace android;
 
+
```
Comment (m4xw) [inline source/main.cpp]: Remove \n

File: source/main.cpp [hunk 9]
```diff
@@ -29,8 +29,66 @@
...
+    usbExit();
+    hidExit();
+    fsExit();
+    smExit();
+}
+#endif // WANT_SYSMODULE
+
 static void stop_thread(MtpServer* server)
 {
+    #ifdef WANT_APPLET
```
Comment (m4xw) [inline source/main.cpp]: remove tab

File: source/main.cpp [hunk 10]
```diff
@@ -42,6 +100,11 @@ static void stop_thread(MtpServer* server)
             break;
         }
     }
+    #endif // WANT_APPLET
```
Comment (m4xw) [inline source/main.cpp]: remove tab

File: source/main.cpp [hunk 11]
```diff
@@ -42,6 +100,11 @@ static void stop_thread(MtpServer* server)
...
+
+    #ifdef WANT_SYSMODULE
```
Comment (m4xw) [inline source/main.cpp]: remove tab

File: source/main.cpp [hunk 12]
```diff
@@ -42,6 +100,11 @@ static void stop_thread(MtpServer* server)
...
+    serverExit = server;
+    #endif // WANT_SYSMODULE
```
Comment (m4xw) [inline source/main.cpp]: remove tab

File: source/main.cpp [hunk 13]
```diff
@@ -71,8 +134,10 @@ int main(int argc, char* argv[])
         }
     }
 
+#ifdef WANT_APPLET
```
Comment (m4xw) [inline source/main.cpp]: remove tab

**Review by p-sam — APPROVED** [review]

**Pull request merged**